### PR TITLE
Added jsdoc

### PR DESCRIPTION
**Pull request by Aymene-n**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 1]
```diff
@@ -566,6 +566,20 @@ const updateRecord = async (record, recordFromTable) => {
   }
 };
 
+/**
+ * Updates a record in a specified DynamoDB table.
+ *
+ * @async
+ * @function updateRecordInTable
+ * @param {Object} client - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:574]: I believe it was requested to change all client definitions to their respective AWS resources

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 2]
```diff
@@ -634,6 +645,17 @@ export async function updateRecordInTable(
   }
 }
 
+/**
+ * Overwrites a record in a specified DynamoDB table.
+ *
+ * @async
+ * @function overwriteRecordInTable
+ * @param {Object} client - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:653]: same here

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 3]
```diff
@@ -654,6 +676,16 @@ export async function overwriteRecordInTable(
   }
 }
 
+/**
+ * Deletes a record from a specified DynamoDB table.
+ *
+ * @async
+ * @function deleteTableRecord
+ * @param {Object} client - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:684]: and here

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 4]
```diff
@@ -708,6 +756,17 @@ function formatEpisodeDeleteItem(table, record) {
   return input;
 }
 
+/**
+ * Puts a new record into a specified DynamoDB table.
+ *
+ * @async
+ * @function putTableRecord
+ * @param {Object} client - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:764]: here too

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 5]
```diff
@@ -813,8 +888,15 @@ function formatEpisodePutItem(table, newRecord) {
...
 
-// returns LSOA using patient postcode or the LSOA from GP practice
-// or returns error message
+/**
+ * Returns the LSOA code using patient postcode or the LSOA from GP practice, or returns an error message.
+ *
+ * @async
+ * @function getLsoa
+ * @param {Object} record - The record containing postcode and primary care provider.
+ * @param {Object} dbClient - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:897]: here as well

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 6]
```diff
@@ -886,7 +974,16 @@ export const filterUniqueEntries = (cassFeed) => {
...
 
-// S3 FUNCTIONS
+/**
+ * Reads a CSV file from an S3 bucket.
+ *
+ * @async
+ * @function readCsvFromS3
+ * @param {string} bucketName - The S3 bucket name.
+ * @param {string} key - The key of the CSV file.
+ * @param {Object} client - The S3 client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:984]: here

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 7]
```diff
@@ -903,6 +1000,17 @@ export const readCsvFromS3 = async (bucketName, key, client) => {
...
 
+/**
+ * Uploads a CSV file to an S3 bucket.
+ *
+ * @async
+ * @function pushCsvToS3
+ * @param {string} bucketName - The S3 bucket name.
+ * @param {string} key - The key of the CSV file.
+ * @param {string} body - The CSV file content.
+ * @param {Object} client - The S3 client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:1011]: here

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 8]
```diff
@@ -941,16 +1057,30 @@ export const parseCsvToArray = async (csvString) => {
...
 };
 
-// DYNAMODB FUNCTIONS
-// returns successful response if attribute doesn't exist in table
+/**
+ * Looks up an item in a DynamoDB table.
+ *
+ * @async
+ * @function lookUp
+ * @param {Object} dbClient - The DynamoDB client.
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:1080]: here as well and the one below

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js
Comment (chenglawson1) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:1]: Missing jsdoc for these functions:
generateInvitesTriggerLambda
mergeRecords 
updateRecord

File: terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js [hunk 9]
```diff
@@ -869,7 +951,13 @@ export const getLsoa = async (record, dbClient) => {
   }
 };
 
-// returns unique records array and an array that contains all the duplicate
+/**
+ * Filters unique entries from a given array and returns unique and duplicate arrays.
+ *
+ * @function filterUniqueEntries
+ * @param {Array} cassFeed - The array to filter.
+ * @returns {Array} An array containing unique records array and an array of duplicates.
```
Comment (chenglawson1) [inline terraform/src/caasFeedUpdateRecordsLambda/caasFeedUpdateRecordsLambda.js:959]: Could change return type to Array<Array>.

**Review by chenglawson1 — APPROVED** [review]

**Review by Amar-NHS — APPROVED** [review]

**Review by chenglawson1 — APPROVED** [review]

**Review by Amar-NHS — APPROVED** [review]

**Pull request merged**